messaging/src/test/java/com/iluwatar/messaging/controller/OrderControllerIntegrationTest.java

```diff
@@ -0,0 +1,117 @@
+package com.iluwatar.messaging.controller;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.iluwatar.messaging.model.MenuItemIdAndQuantity;
+import com.iluwatar.messaging.model.OrderRequest;
+import jakarta.servlet.ServletContext;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.kafka.test.context.EmbeddedKafka;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.RequestBuilder;
+import org.springframework.test.web.servlet.ResultActions;
+
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@ExtendWith(SpringExtension.class)
+@SpringBootTest
+@AutoConfigureMockMvc
+@DirtiesContext
+@EmbeddedKafka(partitions = 1, brokerProperties = {"listeners=PLAINTEXT://localhost:9092", "port=9092"})
+public class OrderControllerIntegrationTest {
+
+  ObjectMapper objectMapper = new ObjectMapper();
+  @Autowired
+  private MockMvc mockMvc;
+
+  @Test
+  void shouldPlaceTheOrderAndReturn_200WhenOrderIsOk() throws Exception {
+
+    OrderRequest orderRequest = new OrderRequest();
+    orderRequest.setConsumerId(1001L);
+    orderRequest.setRestaurantId(2001L);
+    orderRequest.setMenuItemIdAndQuantityList(testMenuItems());
```

Comment:
Can we set these properties in the constructor?

Reply:
updated with constructor